### ISSUE TITLE
Timeshift backs up its own snapshots recursively

### ISSUE THREAD

**Issue by pgmer6809** [teejee2008/timeshift#312]
**Is your feature request related to a problem? Please describe.**
I setup Timeshift to make snapshots to /Backup
It worked the first time. But the second time I ran it, the second snapshot included all of the contents of the first snapshot. (see below),
Perhaps this is because I had mounted /Backup via fstab at boot time? 
And I did not create an exclude filter to exclude it. 

**Describe the solution you'd like**
Timeshift should be smart enought not to include its own previous snapshots in the current snapshot. Perhaps by having a default filter to exclude these, perhaps by some other means. 
Also a prompt to the user to make his own filter if nothing else.

**Describe alternatives you've considered**

**Additional context**

```
/Backup/timeshift/snapshots$ ls
2018-09-12_22-46-58  2018-09-20_17-59-14
```

> As you see there are two directories in the /Backup/timeshift/snapshots directory. An old one, and the one I created today. 
> here is the  tree cmd.  I ran on the new snapshot 

**> `/Backup/timeshift/snapshots$` tree -L 7 2018-09-20_17-59-14 >~/Pictures/tree-timeshift.txt`
> And here is the result. 
> You will see that the old snapshot has been copied completely into the new snapshot.
> In fact even the new snapshot is in the new snapshot.....
> ------------------------------------------------------------------------**

> 2018-09-20_17-59-14

├── exclude.list
├── localhost
│   ├── Backup
│   │   ├── lost+found [error opening dir]
│   │   ├── timeshift
│   │   │   ├── snapshots
│   │   │   │   **├── 2018-09-12_22-46-58**
│   │   │   │   │   ├── exclude.list
│   │   │   │   │   ├── info.json
│   │   │   │   │   ├── localhost
│   │   │   │   │   │   ├── Backup
│   │   │   │   │   │   ├── bin
│   │   │   │   │   │   ├── boot
│   │   │   │   │   │   ├── cdrom
│   │   │   │   │   │   ├── Data
│   │   │   │   │   │   ├── dev
│   │   │   │   │   │   ├── etc
│   │   │   │   │   │   ├── home
│   │   │   │   │   │   ├── initrd.img -> boot/initrd.img-4.15.0-20-generic
│   │   │   │   │   │   ├── initrd.img.old -> boot/initrd.img-4.15.0-20-generic
│   │   │   │   │   │   ├── lib
│   │   │   │   │   │   ├── lib64
│   │   │   │   │   │   ├── media
│   │   │   │   │   │   ├── mnt
│   │   │   │   │   │   ├── opt
│   │   │   │   │   │   ├── proc
│   │   │   │   │   │   ├── root
│   │   │   │   │   │   ├── run
│   │   │   │   │   │   ├── sbin
│   │   │   │   │   │   ├── srv
│   │   │   │   │   │   ├── sys
│   │   │   │   │   │   ├── tmp
│   │   │   │   │   │   ├── usr
│   │   │   │   │   │   ├── var
│   │   │   │   │   │   └── vmlinuz -> boot/vmlinuz-4.15.0-20-generic
│   │   │   │   │   ├── rsync-log
│   │   │   │   │   └── rsync-log-changes
│   │   │   │   └── **2018-09-20_17-59-14**
│   │   │   │       ├── exclude.list
│   │   │   │       ├── localhost
│   │   │   │       │   ├── Backup
│   │   │   │       │   ├── bin
│   │   │   │       │   ├── boot
│   │   │   │       │   ├── cdrom
│   │   │   │       │   ├── Data
│   │   │   │       │   ├── dev
│   │   │   │       │   ├── etc
│   │   │   │       │   ├── home
│   │   │   │       │   ├── initrd.img -> boot/initrd.img-4.15.0-34-generic
│   │   │   │       │   ├── initrd.img.old -> boot/initrd.img-4.15.0-20-generic
│   │   │   │       │   ├── lib
│   │   │   │       │   ├── lib64
│   │   │   │       │   ├── media
│   │   │   │       │   ├── mnt
│   │   │   │       │   ├── opt
│   │   │   │       │   ├── proc
│   │   │   │       │   ├── root
│   │   │   │       │   ├── run
│   │   │   │       │   ├── sbin
│   │   │   │       │   ├── srv
│   │   │   │       │   ├── sys
│   │   │   │       │   ├── tmp
│   │   │   │       │   ├── usr
│   │   │   │       │   ├── var
│   │   │   │       │   ├── vmlinuz -> boot/vmlinuz-4.15.0-34-generic
│   │   │   │       │   └── vmlinuz.old -> boot/vmlinuz-4.15.0-20-generic
│   │   │   │       └── rsync-log
│   │   │   ├── snapshots-boot
│   │   │   ├── snapshots-daily
│   │   │   ├── snapshots-hourly
│   │   │   ├── snapshots-monthly
│   │   │   ├── snapshots-ondemand
│   │   │   │   └── 2018-09-12_22-46-58 -> ../snapshots/2018-09-12_22-46-58
│   │   │   └── snapshots-weekly
│   │   │       └── 2018-09-12_22-46-58 -> ../snapshots/2018-09-12_22-46-58
│   │   └── TImeshift-Summary-20180912.txt
│   ├── bin [error opening dir]
│  _<snip ...,. lines deleted ..>_
├── rsync-log
└── rsync-log-changes

82 directories, 21 

> files

***This is a free application created in my spare time. Consider making a donation to sponsor the work for your request.***

[![](https://www.paypalobjects.com/en_US/i/btn/btn_donate_LG.gif)](https://www.paypal.com/cgi-bin/webscr?business=teejeetech@gmail.com&cmd=_xclick&currency_code=USD&item_name=Timeshift%20Donation)

**Comment by mdrossmiller** [comment]
Yeah, running 19.08; just did this to myself; had the drive for snapshots mounted under root, and didn't think to exclude it , filled a 500G drive in no time; it just kept infinite recursively adding files to the active snapshot. 

Not a big deal to add an exclude, but it would have been nice if the condition had been detected as an error in the config, rather than filling up the disk.
